master/buildbot/config.py

```diff
@@ -818,6 +819,10 @@ def __init__(self, name=None, slavename=None, startSlavenames=None, slavenames=N
         self.tags = tags
         self.build_tags = build_tags
 
+        # Additional properties that must match to allow buildrequests of
+        # this config to be merged
+        self.mergeProperties = mergeProperties or []
```

Comment:
I think this introduces a new way of checking for merges that might not be compatible with the merging functions (for example this could break partition build request for parallel tests where a builder may or not have the partition id, one example is  "Test IntegrationTests - Windows64Editor" builder

I think best case we could try to reuse the logic here: FindPreviousSuccessBuildMixin

Reply:
All existing merging functions use the MatchProperty function, which basically compares properties and ensure they are the same. It would be easy to reproduce that using this list of property names and will not need any python function calls.

Reply:
about parallel test and self.mergeProperties, the default function don't have the partitionid property and it is created dynamically, I’m assuming that the mergeProperty list is fixed in the config and maybe not considering this case? so non-partition builds could potentially get merged

Reply:
> I’m assuming that the mergeProperty list is fixed in the config

yes

> so non-partition builds could potentially get merged

I don't see why that would happen. If you are a partitionable config you will specifically say that we must check `partition-id` property to merge. That means that parition 1s only get merged to partition 1s. If you happen to run a partitionable config without partitioning it (do we even allow that?) then your property value will be `None`, and you will only merge against other identical builds.

Reply:
I'll include @tdctaz to help check this part, I think the problem could be "Test IntegrationTests - Windows64Editor" builder (with no partition set by default) tries to merge with "Test IntegrationTests - Windows64Editor" builder" (that has the extra partition property set by trigger scheduler https://github.com/Unity-Technologies/katana/blob/katana/master/buildbot/steps/partition_trigger.py#L43)

Reply:
So the top level build will have partition `None`, and the child builds will have partition `1,2,3,etc`.

I'm assuming that the child builds will have their `partition` property set when they are added to the queue, or is that only set when they start running?

Reply:
yes the `partition` property is set when added to the queue I guess if we set in the configuration for the parallel builders to compare `partitions-id` then it could work. But I would like to check this also with @tdctaz

Reply:
I think this is fine, but read my general comments below - I think the artifact merge should also be updated and possible the old merge logic should be removed so there is only one merge system active.